internal/mode/static/nginx/runtime/manager_test.go

```diff
@@ -23,6 +23,42 @@ var _ = Describe("NGINX Runtime Manager", func() {
 	})
 })
 
+func TestEnsureNginxRunning(t *testing.T) {
+	ctx := context.Background()
+	cancellingCtx, cancel := context.WithCancel(ctx)
+	time.AfterFunc(1*time.Millisecond, cancel)
```

Comment:
You should just be able to use `context.WithTimeout`.

Reply:
will add an update here